chore: support modal only for cloud

tiina303 opened the pull request:
## Problem

<!-- Who are we building for, what are their needs, why is this important? -->
https://posthog.slack.com/archives/C0460J93NBU/p1686732765130639?thread_ts=1686731931.713799&cid=C0460J93NBU

## Changes

<!-- If there are frontend changes, please include screenshots. -->
<!-- If a reference design was involved, include a link to the relevant Figma frame! -->

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

<!-- Briefly describe the steps you took. -->
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

tiina303 committed:
chore: support modal only for cloud

Twixes reviewed (changes requested):
Definitely makes sense to hide this on self-hosted, but the way this PR currently works, you can still click "Report a bug", it's just that nothing happens on self-hosted. This looks like, well, a bug. So I'm here to report it. :D
I don't think we need to change anything in `SupportModal` in this PR. Instead, we should hide the reporting buttons themselves in `HelpButton`.

tiina303 commented:
I disagree about DEBUG - I don't want folks to be able to submit Zendesk issues in debug mode - but I can move that to be around Zendesk sending part and error out saying it's only supported on cloud. The problem with the menu option is that the button / field exists in other places too.

tiina303 committed:
add error toast

Twixes commented on frontend/src/lib/components/Support/SupportModal.tsx:
```diff
@@ -45,6 +45,12 @@ export function SupportModal({ loggedIn = true }: { loggedIn?: boolean }): JSX.E
 
+    if (!preflightLogic.values.preflight?.cloud) {
+        if (isSupportFormOpen) {
+            lemonToast.error(`Support tickets are only supported on PostHog cloud`)
```
We can just hide the whole support section on self-hosted, since we explicitly don't offer support for hobby setups. In terms of UX, there should _never_ be any button in an interface whose only function is showing an error toast

tiina303 replied:
Sure, but as a backup I want us to block folks from being able to send requests if we fail to exclude the button in some of the locations.

Twixes replied:
That's fair, but the primary strategy should still be hiding the support buttons/links themselves. This PR doesn't do that.

tiina303 replied:
will do a separate PR for that as it could be controversial what exactly we hide

Twixes committed:
Reword toast message

Just wanted it to be a bit less like "use PostHog Cloud" and more "we just don't offer support for self-hosted"

Twixes reviewed (approved):
Makes sense with #16117. I changed the words a bit, let me know if this isn't right

tiina303 commented on frontend/src/lib/components/Support/SupportModal.tsx:
```diff
@@ -40,11 +40,17 @@ const SUPPORT_TICKET_KIND_TO_PROMPT: Record<SupportTicketKind, string> = {
 
+    if (!preflightLogic.values.preflight?.cloud) {
+        if (isSupportFormOpen) {
+            lemonToast.error(`Support via modal isn't provided for self-hosted instances.`)
```
nit I'd say support form or support tickets but 🚢

Twixes replied:
Hmm, it _was_ a bit clunky, I think the most to-the-point naming for this is gonna be "in-app support"

Twixes committed:
Reword error
